Sources/NDT7Test.swift

```diff
@@ -138,20 +138,23 @@ extension NDT7Test {
     /// - parameter completion: closure for callback.
     /// - parameter error: returns an error if exist.
     func serverSetup<T: URLSessionNDT7>(session: T = URLSession.shared as! T,
-                                        useNDT7ServerCache: Bool = false,
                                         _ completion: @escaping (_ error: NSError?) -> Void) {
-        guard settings.url.hostname.isEmpty else {
-            completion(nil)
-            return
-        }
-        discoverServerTask = NDT7Server.discover(session: session,
-                                                 withGeoOptions: settings.useGeoOptions,
-                                                 retray: 4,
-                                                 geoOptionsChangeInRetray: true,
-                                                 useNDT7ServerCache: useNDT7ServerCache, { [weak self] (server, error) in
+        discoverServerTask = NDT7ServerV2.discoverV2(session: session, { [weak self] (server, error) in
             guard let strongSelf = self else { return }
-            strongSelf.settings.url.hostname = server?.fqdn ?? ""
-            strongSelf.settings.url.server = server
+            guard error == nil else { completion(error); return }
+            guard let server = server, server.count != 0 else {
+                let setupError = NSError(domain: NDT7WebSocketConstants.domain, code: 0,                                       userInfo: [ NSLocalizedDescriptionKey: "Failed to locate a valid mlab server to contact"])
+                completion(setupError)
+                return
+            }
+            
+            // Save all server options, load the first server's information in to try to use it first
+            strongSelf.settings.allServers = server
+            strongSelf.settings.currentServerIndex = 0
+            strongSelf.settings.hostname = server[0].machine
+            strongSelf.settings.downloadUrl = server[0].urls.downloadUrl
+            strongSelf.settings.uploadUrl = server[0].urls.uploadUrl
```

Comment:
you have already set the currentServerIndex to 0, so, you should be able to get machine and urls from the servers instead to use these parameters.

Reply:
Yes. That was the original intent, we ended up shipping this version instead because originally this had to be out by Nov 4.